### PR TITLE
Run "make test-style" as part of CI

### LINKED ISSUE #1122: Static analysis and linters should be run as part of CI
We want to run `gometalinter` as part of our CI pipeline.

### PR DESCRIPTION
**Pull request by seanknox**
**What this PR does / why we need it**: We want to run `make test-style` as part of our CI. This PR fixes go style errors and static analysis errors and adds `make test-style` to `make ci`.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1122
**Special notes for your reviewer**:

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1123)

### CODE REVIEW

File: test/acs-engine-test/config/config.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ type Deployment struct {
 	SkipValidation    bool   `json:"skip_validation,omitempty"`
 }
 
+// TestConfig represents a cluster config
```
Comment (seanknox) [inline test/acs-engine-test/config/config.go]: @dmitsh I did my best document the go funcs under test/. Can you double check? Will call out some that I wasn't able to infer about.

File: test/acs-engine-test/report/report.go [hunk 2]
```diff
@@ -111,7 +121,8 @@ func New(jobName string, buildNum int, nDeploys int) *ReportMgr {
 	return h
 }
 
-func (h *ReportMgr) Copy() *ReportMgr {
+// TBD
+func (h *Manager) Copy() *Manager {
```
Comment (seanknox) [inline test/acs-engine-test/report/report.go]: @dmitsh needs definition

File: test/acs-engine-test/report/report.go [hunk 3]
```diff
@@ -125,7 +136,8 @@ func (h *ReportMgr) Copy() *ReportMgr {
 	return n
 }
 
-func (h *ReportMgr) Process(txt, testName, location string) *ErrorInfo {
+// TBD
+func (h *Manager) Process(txt, testName, location string) *ErrorInfo {
```
Comment (seanknox) [inline test/acs-engine-test/report/report.go]: @dmitsh needs function definition

File: test/acs-engine-test/report/report.go [hunk 4]
```diff
@@ -136,7 +148,7 @@ func (h *ReportMgr) Process(txt, testName, location string) *ErrorInfo {
 	return NewErrorInfo(testName, ErrUnknown, ErrClassNone, location)
 }
 
-func (h *ReportMgr) addFailure(key string, locations map[string]int) {
+func (h *Manager) addFailure(key string, locations map[string]int) {
```
Comment (seanknox) [inline test/acs-engine-test/report/report.go]: @dmitsh needs function definition

File: test/acs-engine-test/report/report.go [hunk 5]
```diff
@@ -162,7 +174,8 @@ func (h *ReportMgr) addFailure(key string, locations map[string]int) {
 	h.Errors += cnt
 }
 
-func (h *ReportMgr) CreateTestReport(filepath string) error {
+// TBD
```
Comment (seanknox) [inline test/acs-engine-test/report/report.go]: @dmitsh needs function definition

File: test/acs-engine-test/report/report.go [hunk 6]
```diff
@@ -180,7 +193,8 @@ func (h *ReportMgr) CreateTestReport(filepath string) error {
 	return nil
 }
 
-func (h *ReportMgr) CreateCombinedReport(filepath, testReportFname string) error {
+// TBD
```
Comment (seanknox) [inline test/acs-engine-test/report/report.go]: @dmitsh needs function definition

**Comment by seanknox** [comment]
Two things left for this PR:
1. ignore `pkg/acsengine/templates.go` in linter checks 
1. document outstanding go functions under `test/` (@dmitsh)

Looking at options for number 1...

**Comment by seanknox** [comment]
@acs-bot test this please

**Comment by JackQuincy** [comment]
I like this a lot, though I'm not sure what to do about templates.go. Can we just run it before we generate templates.go? Or does it require  complied package files to run?

**Comment by JackQuincy** [comment]
just saw something the hidden/dot files are ignored in gofmt. Possibly we could just name it .template.go

**Comment by seanknox** [comment]
> I like this a lot, though I'm not sure what to do about templates.go. Can we just run it before we generate templates.go? Or does it require complied package files to run? Though that isn't a good answer for the dev. env Is there a way to add ignored files for gofmt similar to git ignore?

That's what I did for now. Also like renaming it to be `.templates.go`! Let me try that...

**Comment by seanknox** [comment]
> just saw something the hidden/dot files are ignored in gofmt. Possibly we could just name it .template.go

@JackQuincy that works with gofmt! Nice. What other changes are needed so `engine.go` reads in `.template.go`? Haven't been able to figure that out.

```
$ make build
go generate -v `go list ./... | grep -v "github.com/Azure/acs-engine/vendor" | sed 's|github.com/Azure/acs-engine|.|g' | grep -v -w '^.$'`
...
pkg/acsengine/azureconst.go
pkg/acsengine/const.go
pkg/acsengine/defaults.go
pkg/acsengine/doc.go
pkg/acsengine/engine.go
pkg/acsengine/engine_test.go
pkg/acsengine/fileloader.go
.templates.go
...
GOBIN=/Users/seknox/src/go/src/github.com/Azure/acs-engine/bin go install  -ldflags ' -X github.com/Azure/acs-engine/cmd.BuildSHA=3bc9393adb2cd09badc42c08dedfda9c41f9dd01 -X github.com/Azure/acs-engine/cmd.BuildTime=2017-07-28T11:04:34-0700'
# github.com/Azure/acs-engine/pkg/acsengine
pkg/acsengine/engine.go:187: undefined: Asset
pkg/acsengine/engine.go:235: undefined: Asset
pkg/acsengine/engine.go:289: undefined: Asset
pkg/acsengine/engine.go:1182: undefined: Asset
pkg/acsengine/engine.go:1215: undefined: Asset
pkg/acsengine/engine.go:1237: undefined: Asset
pkg/acsengine/engine.go:1263: undefined: Asset
pkg/acsengine/engine.go:1303: undefined: Asset
make: *** [build] Error 2
```

**Comment by seanknox** [comment]
@acs-bot test this please

**Comment by JackQuincy** [comment]
If it doesn't just work maybe that won't work for us...

**Comment by seanknox** [comment]
@acs-bot test this please

**Comment by seanknox** [comment]
@acs-bot test this please

**Pull request merged**